Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/Bluetooth.cs

```diff
@@ -72,28 +69,47 @@ public async Task StopSearch()
             }
         }
         
-        public async Task StartRssiPolling(String btguid, Func<int, int> updateRssi)
+        public void StartRssiPolling(String btguid, Action<int> updateRssi, Action connected = null, Action disconnected = null)
         {
-            try
+            StopRssiPolling();
+            rssiCancel = new CancellationTokenSource();
+            var token = rssiCancel.Token;
+            Task.Run(async () =>
             {
-                IDevice device = await adapter.ConnectToKnownDeviceAsync(Guid.Parse(btguid));
-
-                rssiPollingTimer = new Timer(async (o) => {
-
-                    await device.UpdateRssiAsync();
-                    updateRssi.Invoke(device.Rssi);
-
-                }, "", 0, RssiPollInterval);
-            }
-            catch (Exception e)
-            {
-                Console.WriteLine(e.ToString());
-            }
+                while (!token.IsCancellationRequested)
+                {
+                    try
+                    {
+                        IDevice device = await adapter.ConnectToKnownDeviceAsync(Guid.Parse(btguid));
```

Comment:
Are we sure an exception is thrown, when the device could not be found?
If not, we would repeatedly invoke `disconnect`. Ok for now, we have to keep that in mind for the future.

Reply:
@NicoStellwag and I checked it. An exception is thrown; good behaviour.